bay/docker/towline.py

```diff
@@ -26,9 +27,16 @@ def _read_file(self, path, default=None):
         """
         try:
             tar_stream = self.host.client.get_archive(self.container_name, path)[0]
-            tar = tarfile.open(fileobj=BytesIO(tar_stream.read()))
-            contents = tar.extractfile(tar.getmembers()[0]).read().strip()
-            return contents or default
+            destination = tempfile.NamedTemporaryFile()
```

Comment:
@venkatesh-eb could you verify the untar logic now that `get_archive` returns a generator? it looks very verbose, I don't know if there is a simpler way